pkgs/applications/editors/pulsar/default.nix

```diff
@@ -0,0 +1,180 @@
+    runHook preInstall
+
+    mkdir -p $out/opt/Pulsar
+    mv * $out/opt/Pulsar
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      # needed for gio executable to be able to delete files
+      --prefix "PATH" : "${lib.makeBinPath [ glib ]}"
+    )
+  '' + lib.optionalString (useHunspell && stdenv.hostPlatform.system == "x86_64-linux") ''
+    opt=$out/opt/Pulsar
+    # We need to patch the already existing app.asar.unpacked, and add python3
+    # to resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/node_gyp_bins/python3
+    rm $opt/resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/node_gyp_bins/python3
+    ln -s ${python3}/bin/python3 $opt/resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/node_gyp_bins/python3
+
+    # We need to extract the app.asar file
+    asar extract $opt/resources/app.asar ./app
+
+    # Remove lines 114 to 116 (inclusive) and replace with ''${localePaths}
+    sed -i '114,116d' ./app/node_modules/spell-check/lib/locale-checker.js
+    echo 'sed -i "114i ${localePatchs}" ./app/node_modules/spell-check/lib/locale-checker.js'
+    sed -i "114i ${localePatchs}" ./app/node_modules/spell-check/lib/locale-checker.js
+    # Rebuild app.asar and clean up
+    asar pack ./app $opt/resources/app.asar
+    rm -rf ./app
+  '' + lib.optionalString (useHunspell && stdenv.hostPlatform.system == "aarch64-linux") ''
+    # On aarch64, we must inject our dictionnaries
+    ${hunspellCopyCommands}
+  '';
+
+  postFixup = ''
+    opt=$out/opt/Pulsar
+    # Patch the prebuilt binaries
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${newLibpath}:$opt" \
+      --add-needed libffmpeg.so \
+      --add-needed libxshmfence.so.1 \
+      --add-needed libxkbcommon.so.0 \
+      --add-needed libxkbfile.so.1 \
+      --add-needed libsecret-1.so.0 \
+      $opt/pulsar
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${newLibpath}" \
+      $opt/resources/app/ppm/bin/node
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      $opt/resources/app.asar.unpacked/node_modules/symbols-view/vendor/ctags-linux
+
+  '' + lib.optionalString (stdenv.hostPlatform.system == "x86_64-linux") ''
+    # Replace the bundled git with the one from nixpkgs
+    dugite=$opt/resources/app.asar.unpacked/node_modules/dugite
+    rm -f $dugite/git/bin/git
+    ln -s ${git}/bin/git $dugite/git/bin/git
+    rm -f $dugite/git/libexec/git-core/git
+    ln -s ${git}/bin/git $dugite/git/libexec/git-core/git
+  '' + ''
+    # Patch the bundled node executables
+    find $opt -name "*.node" -exec patchelf --set-rpath "${newLibpath}:$opt" {} \;
+
+    # Create a wrapper script for the executable
+    mkdir -p $out/bin
+    cat > $out/bin/${pname} <<EOF
+    #!${runtimeShell}
+    nohup $opt/pulsar --no-sandbox "\$@" > /dev/null 2>&1 &
+    EOF
+    chmod +x $out/bin/${pname}
```

Comment:
Looking at `makeWrapper`, I couldn't see how it could handle the `nohup`, nor the redirection to /dev/null

(The truly best course of action would be to patch then link `${src}/opt/Pulsar/resources/pulsar.sh` which handle all that, but the script makes wild assumptions about respecting the FHS, while essentially boiling down to doing the same thing as this wrapper)

Reply:
The upstream `pulsar.sh` is defined here:
https://github.com/pulsar-edit/pulsar/blob/03eace728058b1ce0f1143d962dca352edaa90f1/pulsar.sh#L4